pkg/util/href.go

```diff
@@ -0,0 +1,109 @@
+package util
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/readium/go-toolkit/pkg/internal/extensions"
+	"golang.org/x/net/idna"
+)
+
+type QueryParameter struct {
+	name  string
+	value string
+}
+
+type HREF struct {
+	href     string
+	baseHref string
+}
+
+func NewHREF(href string, base string) HREF {
+	return HREF{href: href, baseHref: base}
+}
+
+// Returns the normalized string representation for this HREF.
+func (h HREF) String() (string, error) {
```

Comment:
Added the href tests, and actually had to copy some internal go code to get it working, since for some reason go does not expose some of the necessary logic to escape/unescape a path, just a path fragment. See:
- https://github.com/readium/go-toolkit/blob/parser-mvp-implementation/pkg/internal/extensions/escape.go
- https://github.com/readium/go-toolkit/blob/parser-mvp-implementation/pkg/internal/extensions/string.go#L17
- https://github.com/readium/go-toolkit/blob/parser-mvp-implementation/pkg/util/href.go